code/Controllers/HomeController.cs

```diff
@@ -42,24 +43,38 @@ public HomeController(BagsApi bagsApi)
 			{
 				var product = await bagsApi.GetProduct(productId ?? 0);
 
-				ViewData["url"] = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+				model.Url = $"https://bagcupid.com/app/{path.TrimLeadingSlash()}";
 
-				if (product.Images != null && product.Images.Count() > 0)
-					ViewData["image"] = product.Images
+				if (product?.Images != null && product.Images.Count() > 0)
+					model.Image = product.Images
 						.Aggregate((selectedImage, nextImage) => (nextImage.Priority < selectedImage.Priority) ? nextImage : selectedImage)
 						.Large;
 			}
 
-			return View("index");
+			return View("index", model);
 		}
 	}
 
+	public class MetaViewModel
+	{
+		public String Url { get; set; }
+		public String Type { get; set; }
+		public String Title { get; set; }
+		public String Description { get; set; }
+		public String Image { get; set; }
+	}
+
 	public static class Extensions
 	{
 		public static UInt64? TryParseUInt64(this String value)
 		{
 			UInt64 id = 0;
 			return UInt64.TryParse(value, out id) ? id : (UInt64?)null;
 		}
+
+		public static String TrimLeadingSlash(this String value)
+		{
+			return value.TrimStart('/');
+		}
```

Comment:
Style: Recommend deleting this extension function.  I don't think `foo.TrimLeadingSlash` adds any clarity over `foo.TrimStart('/')`, plus the doc comments are all on `.TrimStart` and `TrimStart` is something that future developers are more likely to be familiar with in terms of behavior.  For example, this actually trims *all* leading slashes, e.g., `///app/product` but that isn't clear without the documentation on `TrimStart`.